### ISSUE TITLE
creating new shipment: gettting all trackers. after this commit  2307e16

### ISSUE THREAD

**Issue by aldf** [magento/magento2#19887]
### Preconditions (*)

1. Magento 2.3-develop

### Steps to reproduce (*)

1. After I pulled 2.3-dev brunch to my code, this [commit ](https://github.com/magento/magento2/commit/2307e16105e031c0826d218cf81ffe7949191a0b) effected creating new order shipment.
2. after creating new shipment and exec $_shipment->getAllTracks()
in https://example.com/admin_xxxxx/admin/order_shipment/view/shipment_id/382
after tick Notify Customer by Email and click save
the customer receive email with all my tracking number 
3. also if the order converted to shipment \Magento\Sales\Model\Convert\Order->toShipment($order);
then $shipment->getAllTracks() will return all trackers

### Expected result (*)

1. return trackers if they already exist or empty array if not exist

### Actual result (*)

1. return all tracker in sales_shipment_track table
![image](https://user-images.githubusercontent.com/11527940/50262510-ae4fc600-044c-11e9-8199-59b9cd70c016.png)

### my temporary fix that work
1- in app/code/Magento/Sales/view/frontend/templates/email/shipment/track.phtml
```
<?php $_shipment = $block->getShipment() ?>
// my fix
<?php $_shipment->getTracksCollection()->addFieldToFilter('parent_id',array('eq' =>$_shipment->getId())); ?>
```

2. in Controller/Adminhtml/Order/Pdfshipments.php
```

$convertOrder = $objectManager->create('Magento\Sales\Model\Convert\Order');
$shipment = $convertOrder->toShipment($order);
// fix the issue after convert get all shipment trackers
$shipment->getTracksCollection()->addFieldToFilter('parent_id',array('eq' =>$shipment->getId()));
```

**Comment by magento-engcom-team** [comment]
Hi @aldf. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@aldf do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by aldf** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @aldf. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @aldf, here is your [Magento instance](https://i-19887-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19887-2-3-develop.instances.magento-community.engineering/admin](https://i-19887-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by aldf** [comment]
i created test order
I didn't receive the email
so I can't use this is instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97260` were created

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by Nazar65** [comment]
I'm will work on this

**Comment by sidolov** [comment]
Hi @aldf. Thank you for your report. 
The issue has been fixed in magento/magento2#19981 by @Nazar65 in 2.3-develop branch
Related commit(s):
 - [8048516db1ef7ea0fd046f2cfe4d04cbc72afdd0](https://github.com/magento/magento2/commit/8048516db1ef7ea0fd046f2cfe4d04cbc72afdd0)
 - [9cf1e3ee05ddcf9f182fe1a16ab0e19e95bd70ec](https://github.com/magento/magento2/commit/9cf1e3ee05ddcf9f182fe1a16ab0e19e95bd70ec)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @aldf. Thank you for your report. 
The issue has been fixed in magento/magento2#20184 by @mage2pratik in 2.2-develop branch
Related commit(s):
 - [f8df2dfd8a99202b52480428618f4c33966a7ba1](https://github.com/magento/magento2/commit/f8df2dfd8a99202b52480428618f4c33966a7ba1)
 - [8b59fc0a3870b5af87256e83353f65e8d676986a](https://github.com/magento/magento2/commit/8b59fc0a3870b5af87256e83353f65e8d676986a)

The fix will be available with the upcoming 2.2.8 release.

**Comment by fabianaromagnoli** [comment]
I think you should reopen this issue, because aldf wrote:

> also if the order converted to shipment \Magento\Sales\Model\Convert\Order->toShipment($order);
then $shipment->getAllTracks() will return all trackers

and this is still true in 2.2.11: if you get all tracks using converter, you get a result (all tracks for all orders), but if you use method getTracksCollection on order, you get a differente result (just one track, and this is correct).
Here are the steps to reproduce:
<img width="686" alt="Screenshot 2020-02-26 at 10 32 09" src="https://user-images.githubusercontent.com/23638004/75332172-3cf77880-5884-11ea-8b69-7f02a6f2ed5c.png">
I think that the fix provided with commit f8df2dfd8a99202b52480428618f4c33966a7ba1 is just a workaround in the phtml. But if you use $shipment->getAllTracks() in other places in your code, you still get the issue.
